### PR TITLE
[WAS-580/WAS-575] - Upgraded strimzi Azure

### PR DESCRIPTION
**Pull request by cenkawolfram**
* Upgraded strimzi version from 0.23 to 0.29
* Upgraded kafka version from 2.8.0 to 3.0.0
* Upgraded kubernetes version 1.24 to 1.25
* Removed old ingress controller and replaced with helm

### CODE REVIEW

**Review by delaix — APPROVED** [review]

File: EnvironmentSetup/Azure/Source/kafka/kafka-persistent.yaml [hunk 1]
```diff
@@ -2,10 +2,9 @@ apiVersion: kafka.strimzi.io/v1beta2
 kind: Kafka
 metadata:
   name: kafka-persistent
-  namespace: kafka
```
Comment (hussainadev) [inline EnvironmentSetup/Azure/Source/kafka/kafka-persistent.yaml:5]: Are you aware of the repercussion(s) removing this?

File: EnvironmentSetup/Azure/Source/kafka/kafka-persistent.yaml [hunk 2]
```diff
@@ -31,16 +30,15 @@ spec:
...
+      log.message.format.version: "3.0"
+      inter.broker.protocol.version: "3.0"
     storage:
       type: jbod
       volumes:
       - id: 0
         type: persistent-claim
         size: 10Gi
         deleteClaim: false
-        class: was-azurefile
```
Comment (hussainadev) [inline EnvironmentSetup/Azure/Source/kafka/kafka-persistent.yaml:43]: Which storageclass are you planning to use? What if there is no default? (The deployment will fail)

**Comment by hussainadev** [comment]
This PR is missing a github actions pipeline to validate the PR builds successfully (you want verify / test deployment - you may consider using kind)

File: EnvironmentSetup/Azure/Source/strimzi-kafka-deployment/060-Deployment-strimzi-cluster-operator.yaml [hunk 3]
```diff
@@ -118,7 +116,7 @@ spec:
           resources:
             limits:
               cpu: 1000m
-              memory: 384Mi
+              memory: 2Gi
```
Comment (hussainadev) [inline EnvironmentSetup/Azure/Source/strimzi-kafka-deployment/060-Deployment-strimzi-cluster-operator.yaml:119]: As per strimzi source code / recommendation this value is `memory: 384Mi` - why has it been changed to `2Gi` ? how was this increased value chosen?

**Review by hussainadev — CHANGES_REQUESTED** [review]

File: EnvironmentSetup/Azure/Source/strimzi-kafka-deployment/060-Deployment-strimzi-cluster-operator.yaml (hunk 3 above)
Comment (cenkawolfram) [reply EnvironmentSetup/Azure/Source/strimzi-kafka-deployment/060-Deployment-strimzi-cluster-operator.yaml:119]: For some reason kafka-persistent-zookeeper pods are exited with OOMKilled when initializing and it caused failure of the script. I increased the limits to 1Gi, the result was same. So I increased it to 2Gi.

Yes, in default they're 384Mi in the helm chart. 
https://github.com/strimzi/strimzi-kafka-operator/blob/0.29.0/helm-charts/helm3/strimzi-kafka-operator/values.yaml#L139-L142

**Review by hussainadev — APPROVED** [review]

**Comment by cenkawolfram** [comment]
We faced OOMKilled issue when testing kafka.
```
[(⎈|WAS-aks:kafka)]$ k top pods
NAME                                       CPU(cores)   MEMORY(bytes)   
kafka-bridge-d9958bc9c-ttcl7               113m         208Mi           
kafka-persistent-kafka-0                   404m         706Mi           
kafka-persistent-kafka-1                   736m         870Mi           
kafka-persistent-kafka-2                   385m         716Mi           
kafka-persistent-zookeeper-0               29m          475Mi           
kafka-persistent-zookeeper-1               130m         597Mi           
kafka-persistent-zookeeper-2               57m          462Mi           
strimzi-cluster-operator-d5b5c5bcb-4ntfq   123m         497Mi 
```
Increased limits.memory to 2Gi

**Pull request merged**